frontend/packages/dev-console/src/components/pipelines/pipeline-builder/utils.ts

```diff
@@ -116,7 +118,7 @@ export const convertBuilderFormToPipeline = (
       ...existingPipeline?.spec,
       params: removeEmptyDefaultFromPipelineParams(params),
       resources,
-      tasks: tasks.map((task) => removeEmptyDefaultParams(removeListRunAfters(task, listIds))),
+      tasks: tasks?.map((task) => removeEmptyDefaultParams(removeListRunAfters(task, listIds))),
```

Comment:
Since we are using this function for `YAML -> formData` conversion, I figured there's nothing stopping the user from deleting `tasks` in the YAML. This function shouldn't fail in that case.

Should we keep using this function for the conversion? Or should I create another function that would wrap around this one, and handle these checks there instead?

Reply:
For one, I don't think you can create a Pipeline without the `tasks` property... so that likely will fail anyways.

As for what if the user removes a needed property? I got around this in my HPA form/yaml work by adding a `sanitizeTo` property to the `formContext` ([link](https://github.com/openshift/console/blob/master/frontend/packages/dev-console/src/components/hpa/HPAForm.tsx#L55-L56)) so we could make sure they can't go the YAML, muck up the resource, go back to the Form and not be able to submit because we don't have a form element for the field they just broke.

Reply:
Ah, you are using it... with this function... okay, so you need to handle that case here. It's important that we respect the types and not just ignore them / change them to suit the need in the moment.

Reply:
I get it now, I looked into the [`sanitizeHPAToForm`](https://github.com/openshift/console/blob/ba3c093702f7784d9da934929977cfa709b4759f/frontend/packages/dev-console/src/components/hpa/hpa-utils.ts#L106) function. I think the important bit is this:
```js
return merge({}, defaultHPA, newFormData);
```
Merging the yaml data with default values, so there is no risk of missing essential fields.
I feel like this might be slightly better, what do you think?

Reply:
Yup, we can definitely take the return of this function, put it in a variable. Then setup a default constant that we use to build the form's base, and merge into an empty object, the default (which has no `tasks`) and then the result from this function.

That _theoretically_ will work. I don't see any issues off the top of my head.

Reply:
Thanks. Updated.